models/classes/render/ContentPostprocessorService.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA
+ */
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\render;
+
+use oat\oatbox\service\ConfigurableService;
+use oat\taoItems\model\render\ItemAssetsReplacement;
+
+class ContentPostprocessorService extends ConfigurableService
+{
+    public function postProcessContent(array $jsonContent): array
+    {
+        $assetService = $this->getAssetsPostprocessor();
+        $jsonAssets = [];
+        if (isset($jsonContent['assets'])) {
```

Comment:
Small suggestion, use early return strategy to avoid the multiple code levels:

```
if (empty($jsonContent['assets'])) {
    return $jsonContent;
}


foreach ($jsonContent['assets'] as $type => $assets) {
....

```